csharp/tools/tracing-config.lua

```diff
@@ -38,13 +39,60 @@ function RegisterExtractorPack(id)
                     match = true
                     break
                 end
+                if arg == 'run' then
+                    -- for `dotnet run`, we need to make sure that `-p:UseSharedCompilation=false` is
+                    -- not passed in as an argument to the program that is run
+                    match = true
+                    needsSeparator = true
+                    injectionIndex = i + 1
+                end
+            end
+            if arg == '--' then
+                needsSeparator = false
+                injectionIndex = i
+                break
             end
         end
         if match then
-            return {
-                order = ORDER_REPLACE,
-                invocation = BuildExtractorInvocation(id, compilerPath, compilerPath, compilerArguments, nil, { '-p:UseSharedCompilation=false' })
-            }
+            local injections = { '-p:UseSharedCompilation=false' }
+            if needsSeparator then
+                table.insert(injections, '--')
+            end
+            if injectionIndex == nil then
+                -- Simple case; just append at the end
+                return {
+                    order = ORDER_REPLACE,
+                    invocation = BuildExtractorInvocation(id, compilerPath, compilerPath, compilerArguments, nil,
+                        injections)
+                }
+            end
+
+            -- Complex case; splice injections into the middle of the command line
+            for i, injectionArg in ipairs(injections) do
+                table.insert(argv, injectionIndex + i - 1, injectionArg)
+            end
+
+            if OperatingSystem == 'windows' then
+                return {
+                    order = ORDER_REPLACE,
+                    invocation = {
+                        path = AbsolutifyExtractorPath(id, compilerPath),
+                        arguments = {
```

Comment:
BuildExtractorInvocation constructs a record line type with the labels `path` and `transformedArguments`. Does that cause problems elsewhere?

Reply:
No, it is only when need to pass in custom arguments on Windows, where we need to use the `commandLineString` field.